components/telemetry-operator/controllers/logpipeline_controller.go

```diff
@@ -242,34 +254,40 @@ func (r *LogPipelineReconciler) syncSectionsConfigMap(ctx context.Context, logPi
 			log.Info("Deleting fluent bit config")
 			delete(cm.Data, cmKey)
 			controllerutil.RemoveFinalizer(logPipeline, sectionsConfigMapFinalizer)
+			changed = true
 		}
 	} else {
 		fluentBitConfig := fluentbit.MergeSectionsConfig(logPipeline)
 		if cm.Data == nil {
 			data := make(map[string]string)
 			data[cmKey] = fluentBitConfig
 			cm.Data = data
-		} else {
-			if oldConfig, hasKey := cm.Data[cmKey]; hasKey && oldConfig == fluentBitConfig {
-				// Nothing changed
-				return false, nil
-			}
+			changed = true
+		} else if oldConfig, hasKey := cm.Data[cmKey]; !hasKey || oldConfig != fluentBitConfig {
 			cm.Data[cmKey] = fluentBitConfig
+			changed = true
+		}
+		if !controllerutil.ContainsFinalizer(logPipeline, sectionsConfigMapFinalizer) {
+			log.Info("Adding finalizer")
+			controllerutil.AddFinalizer(logPipeline, sectionsConfigMapFinalizer)
+			changed = true
 		}
-		controllerutil.AddFinalizer(logPipeline, sectionsConfigMapFinalizer)
 	}
 
 	// Update ConfigMap
-	if err := r.Update(ctx, &cm); err != nil {
-		return false, err
+	if changed {
```

Comment:
May sound picky, since we already have quite a lot of nesting going on (thankfully we have a refactoring issue planned :)
but if you reduce nesting here by `if !changed` just return `false, nil`
as well as in other sync methods